lib.cpp

```diff
@@ -32,6 +37,22 @@ int evaluate(const std::string &s) {
     case '-': change_stack_state(minus,    stack); break;
     case '*': change_stack_state(multiply, stack); break;
     case '/': change_stack_state(divide,   stack); break;
+    case 'l':
+        stack.push('0');
+        change_stack_state(ln, stack);
+        break;
+    case 'e':
+        stack.push('0');
+        change_stack_state(exp,    stack);
```

Comment:
we do now obviously have functions that use two stack variables and functions that only use one.

can you please:

- rename `change_stack_state` to `apply_binary_function(function, stack)`
- add another `apply_unary_function` function and use that in those cases